### DISCUSSION TITLE
Could the app services be put into their own embedded interface?

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#1417] [Q&A]
I'm thinking that it would be nice if the services could be put into their own interface, and the events into another.  This way I could pluck just the services out of `core.App` and to embed it into my Env struct.  This would ensure that events aren't being added anywhere that they shouldn't be.  Also then the UpsertForm could only take a `core.AppServices` interface instead of `core.App`.

https://github.com/pocketbase/pocketbase/blob/master/core/app.go#L19-L93

**Comment by ganigeorgiev** [comment]
Initially `core.App` was split into several different interfaces but at the end I've decided to merge them and use only a single "injection point" since this way it is less likely to introduce a breaking change down the road (for example sometime in the future we may want to trigger/intercept events from inside the forms).

> This would ensure that events aren't being added anywhere that they shouldn't be.

I'm not sure that I understand what do you mean. Even if we remove the hooks from the `core.App`, `app.Dao()` actions for example will still trigger the `OnModel*` events.
If you don't want events, you can create your own instance(s), eg. `dao := daos.New(app.Dao().DB())`.
